### DISCUSSION TITLE
SIMD proposal: Updating Solana inflation rate

### DISCUSSION THREAD

**Discussion by anselsol** [solana-foundation/solana-improvement-documents#200] [SIMD Discussions]
## Summary

The purpose of this feature is to enable Solana reward emission to be dynamic 
on the established inflation schedule parameters.

## Motivation

The inflation rate on Solana should be as low as required to enable more than 
50% of the circulating supply of tokens staked. 

## Detailed Design

The inflation schedule will be implemented on a dynamic curve: 

- the minimum inflation on the curve would be 0%
- the maximum inflation on the curve would be the current Solana curve inflation rate (4.8%)

The inflation shall adjust upwards or downwards at the start of each epoch. 

Upon implementation, inflation would start at 4.8% (this is the current inflation) and:

- inflation would decrease by .1 if the total Solana staked was more than 50%. 
- inflation would increase by .1 if the total Solana staked was less than 50%.

The curve will be bound by :

- current Solana curve inflation rate at the maximum 
- and 0% at the minimum.

## Alternatives Considered

N/A

## Impact

With the evolution of the ecosystem, validators now rely on MEV and priority 
fees over token inflation. This positive direction means that we can reduce 
reliance on inflation for rewarding stakers to maintain network security. 

dApp developers, token holders, and core contributors will have a more sound 
long-term sustainability tokenomics.

## Security Considerations

We want to ensure as many validators as possible are economically sufficient 
to maximize the decentralization of the network.

To ensure this long term, we might need a controller dependent on revenue/opportuniy 
cost rates.

## Simulation
@a2kdefi713 put together this simulation with different deflation scenarii:

https://docs.google.com/spreadsheets/d/1QMPwCbPMRQDfD6864CbrA2wXfmUuzMnvH6blG_IJ5I4/edit?usp=sharing

N/A

**Comment by anselsol** [comment]
![Screenshot 2024-11-22 at 17 09 31](https://github.com/user-attachments/assets/5f542430-60b7-4bac-908f-f886037e52a4)

@levicook that's a very good question. Cosmos has this kind of super complex formula to stabilize oscillations depending on the multiple threshold (not just 50%) - we need to learn more about the implications to be able to reply to this (source : https://forum.cosmos.network/t/proposal-48-accepted-inflation-rate-change-correction/4834)

Right now the curve ([you can find it here](https://solana.com/docs/economics/inflation/inflation-schedule)) goes to 1.5% in 6 years. The idea was to go there much quicker to stabilize the total supply.

From what we could gather a big part of the staking APY do come from fees right now (and the 50% fees burn will soon be removed).

The current inflation is around 4.8%. I think the staking APY (maybe you're talking about this) takes into account auto compounding + base tx fees + priority fees (+ jito tips if running Jito).

**Reply by michaelh-laine** [reply]
> From what we could gather a big part of the staking APY do come from fees right now (and the 50% fees burn will soon be removed).

This is entirely false, 0% of staking APY comes from fees currently and until SIMD-0123 is developed and implemented this will remain that way.

**Comment by fikunmi-ap** [comment]
My primary issue with proposed changes to inflation is that I cannot justify the time spent having these discussions and making changes to the codebase.

Inflation does not affect the security of the network, the only (debatable) benefit to having lower inflation is token health. And while token health is important, we just hit ATHs in spite of the inflation before deflationary ETH. 

Correlation is not always evidence of causation but I can’t help but wonder if there’s any actual value to low inflation.

When I understand your perspective on this, I can make comments on the actual design. It’s mostly good though.

**Reply by a2kdefi713** [reply]
I consider decentralization an important part of network security. Token price is a major consideration for decentralization. Validators have to be able to pay for hardware and labor and they sell SOL to fund their operations. 

Also to reach the ATH it was materially harder to do because of the inflation. As it relates to ETH, the primary challenge they have had is the tech strategy isn’t terribly user friendly - I think if it wasn’t for the deflationary token the trading pair with Solana would have been even worse.

**Reply by fikunmi-ap** [reply]
> Token price is a major consideration for decentralization.

So we should raise inflation so token price is lower?

> Validators have to be able to pay for hardware and labor and they sell SOL to fund their operations.

Lowering or raising inflation doesn't  affect this. What has a more meaningful effect is stability. 

Validator operators will adjust their commission to break even + profit. Any inflation value works for them. Aside tax considerations, this point is moot.

Ironically it's more problematic is if inflation changes every epoch.

> Also to reach the ATH it was materially harder to do because of the inflation.

My (staked) bags already reached ATHs several weeks ago. This pump is icing on the cake for me :)

**Comment by KaneLabs** [comment]
Changing inflation should be completely off the table. If it is changed, or gets taken seriously that it can be changed, it introduces an unquantifiable risk into the long-term financial modeling of the value of the token. What's stopping the token from changing every quarter or month based on the current macro econ conditions - like the fed? As was posted above, the only purpose is token health, which would be self-defeating. Stability is an important feature. The mc is 100b already, it is clearly not holding back the success of the network. I only see this introducing downside risk.

**Reply by a2kdefi713** [reply]
Solana has exceeded expectations on all metrics. Recognizing the success and adjusting is better than continuing to dilute holders and the health of the token. Also, don’t discount the price of the token and the ability to further decentralize the network as validators have to pay for hardware and labor in dollars. 

Material things can change and the world won’t fall apart. Example validators now take 100% of the priority fee instead of burning half. Many senior devs and leaders had previously communicated that once Solana was at 1.5% and TPS with priority fee was high enough that Solana would be deflationary. 

The argument around Stability has been highly subjective.

**Reply by anselsol** [reply]
Just adding this here in case it's not obvious from the SIMD WIP: https://solana.com/docs/economics/inflation/inflation-schedule

The inflation would never be higher than the base inflation curve. Meaning that we could only accelerate toward the goal of 1.5% 6 years in advance (like @a2kdefi713 said, Solana has been sprinting to its goals much faster than expected)

**Comment by nope-finance** [comment]
Is <50% stake participation a goal of solanas?

Additionally I think this makes last slot of epoch even more valuable than it already is as proposed not sure if you have thought about this?

**Reply by anselsol** [reply]
It could be 50% of anything else, I don't think we are looking for having less than 50% of tokens staked. IMO the question is more what is the lowest safest inflation we can have right now? And should we change it at all?

Also the inflation is a just a part of the staking rewards, so people who are staking their SOL aren't going to get a 1.5% APY if the inflation drops to 1.5%

How would the last slot of an epoch be more valuable in this context?

**Reply by nope-finance** [reply]
> How would the last slot of an epoch be more valuable in this context?

the last slot leader can stake or unstake to exactly 50%-1

**Comment by aeyakovenko** [comment]
My two cents:

There isn't enough research to know what the best inflation should be, or what is "safe".  At a gut level, something that is too high is probably wrong, like 100% per epoch would definitely break markets. 

0 could also be wrong and unsafe. While 0 could be unsafe, negative value could be safe to a point. I don't think the right number will provide more security, but the wrong number could definitely break things. 

The current solana curve is going from current value to 1.5% at 15% decrease per year. 

So in some sense we have some blind faith that everything between current value and 1.5% is "safe". We have evidence over the last 3 years that a higher rate is actually not unsafe (nothing broke). 

So it's no less risky than the status quo to let the system float between current value as defined by the curve and 1.5%. 

The next question is - is it unsafe to move inflation faster?  We have a ton of evidence that shows that 15% per year hasn't broken anything. It's possible that going from max to min every epoch would break things. 

Given that inflation started much higher, we actually have some evidence showing that a (8 * 0.15) or 1.2% nominal value change to inflation over a year didn't break anything. So changing the nominal value of inflation by a larger amount than today is probably fine.

It seems unlikely that 30% per year change would break anything.  Possibly unlikely that 45% would either. 

So the way to do this in the safest way possible would be something like:

Max = current curve that is dropping at 15%, so worst case highest inflation rate is the same as the current curve. 

Min = 1.5% same as before, so worse case lowest inflation rate is same as current terminal rate.  

Current = changes between max and min at rate X, where starting value for rate X could be 30% per year or 50% per year.  

The trigger could be 50% stake, or some other arbitrary value or value range to avoid changes every epoch.  The current configuration has a change every epoch, so we have some evidence that small changes to inflation don't break things.  

What's a safe amount of stake is another question. The solana network has operated with less stake than currently and more stake.  I think the lowest was ~20%. Tons of tendermints and tezos have run at > 90%.  Ethereum is at 28% of total. 50% target seems totally fine.

Markets are better than constants.  But don't let the perfect be the enemy of the good.  It's fine to set a long term vision for inflation to be totally market driven, and slowly get there through small incremental changes.  

It's also fine to let things be good enough.  If the perfect is only marginally better but is way too complicated to implement, it's ok to not do it.

**Reply by anselsol** [reply]
> What's a safe amount of stake is another question. The solana network has operated with less stake than currently and more stake. I think the lowest was ~20%. Tons of tendermints and tezos have run at > 90%. Ethereum is at 28% of total. 50% target seems totally fine.

I can't find historical data for this - would be interesting to see if anything changed/peaked or whatever during low staking periods on Solana.

> It's also fine to let things be good enough. If the perfect is only marginally better but is way too complicated to implement, it's ok to not do it.

Right now it makes more sense to focus work on what Solana really needs. We need to figure out if reducing inflation is part of this feature set - based on most people's feedback it doesn't seem so 😅

**Reply by aeyakovenko** [reply]
Do a stake or token weighted poll.

**Reply by anselsol** [reply]
Is there a foundation tool to run such poll or do we need to spin one up for this particular vote?

**Reply by aeyakovenko** [reply]
> Is there a foundation tool to run such poll or do we need to spin one up for this particular vote?

Maybe Jacob Creech has one.  An informal poll with validators would work.

**Reply by jacobcreech** [reply]
@anselsol there's a process outlined for the last validator vote done in https://forum.solana.com/t/proposal-for-enabling-the-reward-full-priority-fee-to-validator-on-solana-mainnet-beta/1456

**Reply by anselsol** [reply]
@jacobcreech awesome thanks I'll take a look at this right now 🤝

**Comment by michaelh-laine** [comment]
My thoughts on this:

- Validators currently derive revenue from up to four sources: commission on inflationary staking rewards, block rewards (incl priority fees), commission on Jito MEV rewards, and side deals (including Paladin P3, SWQOS partnerships, private mempool kickbacks etc)
- Stakers currently derive revenue from up to two sources primarily: inflationary staking rewards and Jito MEV - in a few cases validators share block rewards manually but this is not widespread and not protocolized
- A precusor to any discussion about what to change inflation to is research with empirical evidence as to why the current inflation rate and trajectory are inappropriate as well as guidance as to the degree
- Following such research (I haven't seen any), there needs to be a well-reasoned basis to determine the desired outcomes of the change. This proposal sets the desired stake equilibrium level at 50%, but offers no evidence or justification why this is the optimal level

SImply saying "inflation is bad" or "Validators earn too much" is not evidence and not grounds for a massive economic intervention such as is proposed here. There very well may be justified reasoning why the current inflation schedule is bad, but this reasoning must be surfaced in justification of a change, this is owed to all tokenholders, present and future, who will experience the consequences of this change.
